contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,337 @@
+                IERC20(swapTokens[i]).safeApprove(uniswapAddr, 0);
+            }
+        }
+        uniswapAddr = _address;
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            // Add swap token approvals for new address
+            IERC20(swapTokens[i]).safeApprove(uniswapAddr, type(uint256).max);
+        }
+        emit UniswapUpdated(_address);
+    }
+
+    /**
+     * @dev Add a swap token to the tokens that get liquidated for stablecoins
+     *      whenever swap is called. The token must have a valid feed registered
+     *      with the price provider.
+     * @param _addr Address of the token
+     */
+    function addSwapToken(address _addr) external onlyGovernor {
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            if (swapTokens[i] == _addr) {
+                revert("Swap token already added");
+            }
+        }
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+
+        // Revert if feed does not exist
+        IOracle(priceProvider).price(_addr);
+
+        swapTokens.push(_addr);
+
+        // Give Uniswap infinite approval
+        if (uniswapAddr != address(0)) {
+            IERC20 token = IERC20(_addr);
+            token.safeApprove(uniswapAddr, 0);
+            token.safeApprove(uniswapAddr, type(uint256).max);
+        }
+
+        emit SwapTokenAdded(_addr);
+    }
+
+    /**
+     * @dev Remove a swap token from the tokens that get liquidated for stablecoins.
+     * @param _addr Address of the token
+     */
+    function removeSwapToken(address _addr) external onlyGovernor {
+        uint256 swapTokenIndex = swapTokens.length;
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            if (swapTokens[i] == _addr) {
+                swapTokenIndex = i;
+                break;
+            }
+        }
+
+        require(swapTokenIndex != swapTokens.length, "Swap token not added");
+
+        // Shift everything after the index element by 1
+        for (uint256 i = swapTokenIndex; i < swapTokens.length - 1; i++) {
+            swapTokens[i] = swapTokens[i + 1];
+        }
+        swapTokens.pop();
+
+        if (uniswapAddr != address(0)) {
+            IERC20 token = IERC20(_addr);
+            // Remove Uniswap approval
+            token.safeApprove(uniswapAddr, 0);
+        }
+
+        emit SwapTokenRemoved(_addr);
+    }
+
+    /***************************************
+                    Rewards
+    ****************************************/
+
+    /**
+     * @dev Transfer token to governor. Intended for recovering tokens stuck in
+     *      contract, i.e. mistaken sends.
+     * @param _asset Address for the asset
+     * @param _amount Amount of the asset to transfer
+     */
+    function transferToken(address _asset, uint256 _amount)
+        external
+        onlyGovernor
+    {
+        require(
+            !IVault(vaultAddress).isSupportedAsset(_asset),
```

Comment:
We don't need this check here, since this is not the vault. On the vault we didn't want the governor to be able to steal principal.

Reply:
Oh got it... stealing harvest reward tokens before calling the swap and sending them to the vault + caller is obviously significantly smaller issue.